storage/kv/badger.go

```diff
@@ -231,6 +234,45 @@ func (b *BadgerDb) listRawKeys(prefix []byte, txn *badger.Txn) [][]byte {
 	return keys
 }
 
+// Update is a gateway to badger db Update function
+// creating and managing a read-write transaction
+func (b *BadgerDb) Update(fn func(basedb.Txn) error) error {
+	return b.db.Update(func(txn *badger.Txn) error {
+		return fn(&badgerTxn{txn: txn})
+	})
+}
+
 func isNotFoundError(err error) bool {
 	return err != nil && (err.Error() == "not found" || err.Error() == "Key not found")
 }
+
+type badgerTxn struct {
+	txn *badger.Txn
+}
+
+func (t *badgerTxn) Set(prefix []byte, key []byte, value []byte) error {
+	return t.txn.Set(append(prefix, key...), value)
+}
+
+func (t *badgerTxn) Get(prefix []byte, key []byte) (obj basedb.Obj, found bool, err error) {
```

Comment:
@amir-blox Can we re-use this snippet in `IDb.Get` or would it just complicate or hurt performance? (because of interfaces)

Reply:
Good idea.

`Get` is usually called to fetch a single item,`GetMany` should be called if there are multiple items to read at once (e.g. get decided in range), therefore we shouldn't see a real impact on performance when reusing `Txn.Get`.